mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -563,11 +580,16 @@ class Word2VecModel private[spark] (
       ind += 1
     }
 
+    // NB: This code (and the documented behavior of findSynonyms
```

Comment:
Oh, is the problem that several words, including the word itself, may have 1.0 similarity? and so the word itself may not sort as the top result among the 1.0 results? Yeah, then the code below doesn't quite handle that case. (But below you wouldn't need the `take(num + 1)` now I think?)

What about just ...

```
val scored = wordList.zip(cosVec).toSeq.sortBy(-_._2)
val filtered = wordOpt match {
  Some(word) => scored.take(num + 1).filter(p => word != p._1)
  None => scored
}
filtered.take(num).toArray
```